engine/runtime/src/main/scala/org/enso/interpreter/instrument/job/UpsertVisualisationJob.scala

```diff
@@ -129,7 +129,10 @@ class UpsertVisualisationJob(
     moduleName: String,
     expression: String
   )(implicit ctx: RuntimeContext): Either[EvalFailure, AnyRef] = {
-    val maybeModule = ctx.executionService.findModule(moduleName)
+    val context = ctx.executionService.getContext
+    // TODO [RW] more specific error when the module cannot be installed
```

Comment:
Returning more specific errors would require altering the protocol which is out of scope of this quick fix.

I think that for now the `ModuleNotFound` should be enough as it is correct - the module could not be found, it is just not as specific (whether it could not be resolved or the download failed etc.). I suggest to add this later on.

Reply:
Please file an issue for this.